src/main/java/org/takes/http/BkBasic.java

```diff
@@ -116,11 +117,24 @@ private static Response failure(final Throwable err) {
         writer.close();
         return new RsWithStatus(
             new RsText(new ByteArrayInputStream(baos.toByteArray())),
-            HttpURLConnection.HTTP_INTERNAL_ERROR
+            BkBasic.responseCode(err)
         );
     }
 
     /**
+     * Returns HTTP response code for specified exception.
+     * @param err Exception
+     * @return Numeric HTTP code.
+     */
+    private static int responseCode(final Throwable err) {
+        int result = HttpURLConnection.HTTP_INTERNAL_ERROR;
+        if (err instanceof HttpException) {
```

Comment:
we don't use `instanceof`, it's evil, see http://www.yegor256.com/2015/04/02/class-casting-is-anti-pattern.html